src/dev/build/tasks/clean_tasks.js

```diff
@@ -84,3 +84,28 @@ export const CleanExtraBinScriptsTask = {
     }
   }
 };
+
+export const CleanExtraBrowsersTask = {
+  description: 'Cleaning extra browsers from platform-specific builds',
+
+  async run(config, log, build) {
+    const reporting = 'node_modules/x-pack/plugins/reporting';
+    const browsers = '{.chromium,.phantom}';
+    for (const platform of config.getPlatforms()) {
+      if (platform.isWindows()) {
+        await deleteAll(log, [
+          build.resolvePathForPlatform(platform, reporting, browsers, '*{darwin,macosx,linux}.zip'),
```

Comment:
How about creating a partial function for and just supply the last argument?

Reply:
Or maybe we can create a mapping which would eliminate the duplication?

```
async run(config, log, build) {
  const reportingPath = 'node_modules/x-pack/plugins/reporting';
  const browsers = '{.chromium,.phantom}';

  const builds = ['darwin', 'macosx', 'linux', 'win32', 'windows'];
  const buildMappings = {
    darwin: ['darwin', 'macosx'],
    linux: ['linux'],
    windows: ['win32', 'windows'],
  };

  function browserPathsWithout(platform) {
    return build.resolvePathForPlatform(
      platform,
      reportingPath,
      browsers,
      _.without(builds, buildMappings[platform])
    );
  }

  for (const platform of config.getPlatforms()) {
    await deleteAll(log, [browserPathsWithout(platform)]);
  }
},
```

Reply:
Not sure if that is any more readable though

Reply:
I pushed partials with 70754dd.  I get really confused with negatives/without so I did a positive mapping.